txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -76,16 +76,30 @@ verifyTxUtxo
     :: (MonadUtxoRead m, MonadError ToilVerFailure m)
     => VTxContext
     -> TxAux
-    -> m (TxUndo, TxFee)
+    -> m (TxUndo, Maybe TxFee)
 verifyTxUtxo ctx@VTxContext {..} ta@(TxAux UnsafeTx {..} witnesses _) = do
-    verifyConsistency _txInputs witnesses
-    verResToMonadError (ToilInvalidOutputs . formatFirstError) $
-        verifyOutputs ctx ta
-    resolvedInputs <- mapM resolveInput _txInputs
-    txFee <- verifySums resolvedInputs _txOutputs
-    verifyInputs ctx resolvedInputs ta
-    when vtcVerifyAllIsKnown $ verifyAttributesAreKnown _txAttributes
-    return (map snd resolvedInputs, txFee)
+    let unknownTxInMB = find (isTxInUnknown . snd) $ zip [0..] (toList _txInputs)
+    case (vtcVerifyAllIsKnown, unknownTxInMB) of
+        (True, Just (inpId, txIn)) -> throwError $
+            ToilUnknownInput inpId txIn
+        (False, Just _) -> do
+            -- Case when at least one input isn't known
+            minimalReasonableChecks
+            resolvedInputs <- mapM (runMaybeT . resolveInput) _txInputs
```

Comment:
So what? Again, how does `runMaybeT` help us here?
```
f :: MonadError () m => Bool -> m ()
f True = pure ()
f False = throwError ()

skovoroda :: (MonadBardaq m, MonadError () m) => m ()
skovoroda = do
  kek :: Maybe () <- runMaybeT (f False)
  print kek
  return ()

g :: MonadBardaq m => m ()
g = do
  print =<< runExceptT skovoroda
```

I expect `g` to print `Left ()`. But that's not what you want here.